### PR TITLE
Accept oauth response with code in GET args or JSON params

### PR DESCRIPTION
**Pull request by revmischa**
For client-side auth flows

### CODE REVIEW

**Comment by frol** [comment]
Could you, please, elaborate more on the issue you solve and provide references to the RFC?

**Comment by revmischa** [comment]
I'm doing part of the authorization flow on the client side. The JS client asks the user to auth the application, which then returns the auth code. The client sends the auth code to Flask, which exchanges the code for an access token using the secret on the backend. This is a common way to do oauth2.
Before to make this work I had to overwrite `request.args` and put the code in manually, since it was POSTed. This is the default behavior of libraries like [Satellizer](https://github.com/sahat/satellizer).

**Comment by frol** [comment]
Well, [OAuth2 specification states the following](https://tools.ietf.org/html/rfc6749#section-4.1.1): 

> The client makes a request to the token endpoint by sending the
following parameters using the "application/x-www-form-urlencoded"

Thus, I would avoid implementing this JSON request support by default, but you may adjust the PR so it enables better customization, so you don't need to patch the `request.args`, instead, you just override `authorized_response`.

P.S. It is also a good idea to report the issue to Satellizer if what you describe is still the case since they are off the specification.

**Comment by revmischa** [comment]
I see your point there. The part I am referring to is section "C" of section 4.1  - where the user-agent talks to the client. You are referring to section "D" - where the client talks to the authorization server.

```
   (C)  Assuming the resource owner grants access, the authorization
        server redirects the user-agent back to the client using the
        redirection URI provided earlier (in the request or during
        client registration).  The redirection URI includes an
        authorization code and any local state provided by the client
        earlier.
```
Are there normative guidelines on how this must be performed, or is it up to the user-agent to decide how this takes place? Because this is the part that Satellizer uses JSON for. Yes it's not technically a "redirect", but the user-agent/client request is entirely under the control of the client developer and defining their interaction in a standard would be of little value, since there is no third-party interoperation.
That's just how I read it, I welcome corrections. I agree that revising the patch to make the caller more responsible for where the args come from makes sense.

File: flask_oauthlib/client.py [hunk 1]
```diff
@@ -621,15 +621,15 @@ def handle_oauth1_response(self):
...
         """Handles an oauth2 authorization response."""
 
         client = self.make_client()
         remote_args = {
-            'code': request.args.get('code'),
+            'code': args.get('code'),
             'client_secret': self.consumer_secret,
-            'redirect_uri': session.get('%s_oauthredir' % self.name)
         }
+        remote_args['redirect_uri'] = session.get('%s_oauthredir' % self.name)
```
Comment (frol) [inline flask_oauthlib/client.py]: I cannot quite catch why this change is necessary. Could you, please, shed some light on it?

File: flask_oauthlib/client.py [hunk 2]
```diff
@@ -659,6 +659,7 @@ def handle_oauth2_response(self):
 
         data = parse_response(resp, content, content_type=self.content_type)
         if resp.code not in (200, 201):
+            log.error(data)
```
Comment (frol) [inline flask_oauthlib/client.py]: I think, this log should be dropped.

**Comment by frol** [comment]
@revmischa Thank you for your contribution!
